### PR TITLE
[Explainer] Support Saved Queries in selectURL

### PR DESCRIPTION
**Pull request by pythagoraskitty**
`selectURL()` currently has two per-page-load budgets that restrict that number of calls made to `selectURL()` on each page-load.  We propose allowing queries to be saved and reused on a per-page basis, where the per-page-load budgets are charged the first time a saved query is run but not for subsequent runs of the saved query during the same page-load. This will be accomplished with a `savedQuery` parameter in the options for `selectURL()` that will name the query.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,6 +144,13 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
         *   `keepAlive` (defaults to false), a boolean denoting whether the worklet should be retained after it completes work for this call.
             *   If `keepAlive` is false or not specified, the worklet will shutdown as soon as the operation finishes and subsequent calls to it will fail.
             *   To keep the worklet alive throughout multiple calls to `run()` and/or `selectURL()`, each of those calls must include `keepAlive: true` in the `options` dictionary.
+        *   `savedQuery` (defaults to the empty string), a string naming the query to be saved or reused.
```
Comment (dmdabbs) [inline README.md:147]: The `resolveToConfig` attribute supported in `options` isn't described in this section.

Comment (pythagoraskitty) [reply README.md:147]: Good point, thank you. We will address in a separate PR. Here is an issue to keep track: #177.

File: README.md [hunk 2]
```diff
@@ -148,6 +148,13 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
         *   `resolveToConfig` (defaults to false), a boolean denoting whether the returned promise resolves to a fenced frame config.
             *   If `resolveToConfig` is false or not specified, the returned promise resolves to an opaque URN that can be rendered by an iframe.
             *   If `resolveToConfig` is true, the returned promise resolves to a [fenced frame config](https://github.com/WICG/fenced-frame/blob/master/explainer/fenced_frame_config.md).
+        *   `savedQuery` (defaults to the empty string), a string naming the query to be saved or reused.
+            *   If the value of `savedQuery` is nonempty and has not previously been associated with a [result index](#result-index) for call to `selectURL()` on the same page, and if the call to `selectURL()` succeeds:
+                *    The pair of (`savedQuery`, [`index`](#result-index)) will be stored for the lifetime of the page.
+                *    The shared storage data-owning origin's site can reuse the query from anywhere within the page.
```
Comment (jkarlin) [inline README.md:154]: let's use consistent terminology and call it the data origin.

**Review by jkarlin — APPROVED** [review]
lgtm w/ comment

File: README.md (hunk 2 above)
Comment (pythagoraskitty) [reply README.md]: Done.

**Pull request merged**